### DISCUSSION TITLE
Cloud init config for join nodes

### DISCUSSION THREAD

**Discussion by name212** [kubernetes-sigs/cluster-api#9770] [Q&A] [answered]
Hi all!

We want to use cluster-api in our project. 
Now we pass the initialising script with cloud-init, it downloads a set of scripts from the control-plane node and executes them. As a result, the node is joined in the cluster.
We don't want to stray away from our implementation. Can we use KubeadBootstrapConfig to pass only cloud-init config? So that kubeadm bootstrap provider does not call kubeadm by itself.
Or do we need to implement a new bootstrap provider?

Thanks

**Comment by fabriziopandini** [comment, marked as answer]
the kubeadm bootstrap provider always calls kubeadm (this is why it primarly exists for, providing additional cloud-init settings is a feature to allow some customizations on top of what kubeadm does)
